extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.js

```diff
@@ -86,28 +86,63 @@ ve.init.wikia.ViewPageTarget.static.actionsToolbarConfig = [
 	}
 ];
 
+ve.init.wikia.ViewPageTarget.prototype.onSurfaceReady = function () {
+	// Parent method
+	ve.init.mw.ViewPageTarget.prototype.onSurfaceReady.call( this );
+
+	if ( !this.anonWarningWidget ) {
+		this.anonWarningWidget = new ve.ui.WikiaAnonWarningWidget();
+		this.anonWarningWidget.setupAnonWarning( this.getToolbar() );
+		this.anonWarningWidget.connect( this, {
+			logInSuccess: 'onLogInSuccess'
+		} );
+	}
+
+	if ( !this.licenseWidget ) {
+		this.licenseWidget = new ve.ui.WikiaLicenseWidget();
+		this.licenseWidget.setupLicense( '#WikiaArticle' );
+	}
+
+};
+
 /**
  * @inheritdoc
  */
-ve.init.wikia.ViewPageTarget.prototype.setSurface = function ( surface ) {
+ve.init.wikia.ViewPageTarget.prototype.onSave = function (
+	html, categoriesHtml, newid, isRedirect, displayTitle, lastModified, contentSub
+) {
+	this.tearDownLicense();
```

Comment:
Do we need to do it twice?
Here and in  `tearDownSurface()`?

My intuition is that `tearDownSurface()` should be already called when saving.